app/src/main/java/com/gathering/android/profile/ProfileViewModel.kt

```diff
@@ -0,0 +1,37 @@
+package com.gathering.android.profile
+
+import androidx.lifecycle.ViewModel
+import com.gathering.android.common.ActiveMutableLiveData
+import com.gathering.android.profile.updateinfo.UpdatedUserInfo
+import javax.inject.Inject
+
+class ProfileViewModel @Inject constructor(private val profileRepository: ProfileRepository) :
+    ViewModel() {
+
+    private val _viewState = ActiveMutableLiveData<ProfileViewState>()
+    val viewState: ActiveMutableLiveData<ProfileViewState> by ::_viewState
+
+    fun onViewCreated() {
+        val user = profileRepository.getUserData()
+        _viewState.setValue(ProfileViewState.SetEmail(user.email.toString()))
+        _viewState.setValue(ProfileViewState.ShowImage(user.photoUrl.toString()))
+        _viewState.setValue(ProfileViewState.SetDisplayName(user.displayName.toString()))
+    }
+
+    fun onDisplayNameChanged(displayName: String) {
+        _viewState.setValue(ProfileViewState.SetDisplayName(displayName))
+
+    }
+
+    fun onImageChanged(imgUrl: String) {
+        _viewState.setValue(ProfileViewState.ShowImage(imgUrl))
+    }
+
+    fun onFavoriteEventLayoutClicked() {
+        _viewState.setValue(ProfileViewState.NavigateToFavoriteEvent)
+    }
+
+    fun onPersonalDataLayoutClicked(updatedUserInfo: UpdatedUserInfo) {
+        _viewState.setValue(ProfileViewState.NavigateToPersonalData(updatedUserInfo))
+    }
+}
```

Comment:
correct me if i'm wrong. but I believe we also talked about having a section somewhere in the profile for signout. is sign-out going to be a separate ticket? its perhaps going to be in UpdateUserInfo fragment? since I didn't see it anywhere

Reply:
This is correct, we need such a button in profile page.

Reply:
Actually, I forgot to put the sign_out button. I am going to add the button and its functionality.

Reply:
good idea!